aten/src/ATen/native/cpu/Depthwise.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/native/DispatchStub.h>
+
+/*
+  Caffe 2 depthwise 3x3 convolution operator
+*/
+
+namespace at {
+namespace native {
+
+using convolution_depthwise3x3_winograd_fn =
+    Tensor (*)(const Tensor &, const Tensor &, const Tensor &,IntArrayRef, IntArrayRef, int);
+
+bool is_convolution_depthwise3x3_winograd_implemented();
```

Comment:
constexpr variables and functions are implicitly inline according to the C++ standard, which means their definition must be visible to the compiler within the same compilation unit they are used in. That have brought the #ifdef out of the .cpp file and into the header which is why I decided to make it a run-time check.  The implementation though is conditionally compiled out / in at compile-time depending on whether we have support for the platform.